### ISSUE TITLE
The way judgements work for sliders isn't very good.

### ISSUE THREAD

**Issue by zachllogan** [ppy/osu#3300]
So there is a couple of issues here, so maybe this should be more than one issue, but I figured they are related enough to be under the same one.

My first problem is with slider tails. They give judgements, but these judgements don't affect combo, accuracy, or score (the only reason I can see for them to even exist is to affect the slider completion judgements). The problem I have with this is that it can lead to result screens like this one:
![lazerscorescreenerror](https://user-images.githubusercontent.com/22415468/44505545-8d769000-a667-11e8-9271-0188673fc333.PNG) If I apply the accuracy formula to the the judgement results shown I get something different than the accuracy I achieved because there are extra greats and misses from slider tails that don't affect accuracy. This makes it appear like I got more misses, greats, or both than my accuracy actually reflects. Further, I FCed despite having 2 misses shown, which should not be able to happen. To resolve this you could simply get rid of slider end judgments, or you could make it so they don't count toward the numbers displayed on the results screen.

My second problem is that sliders are incredibly overweighted in terms of accuracy, and their accuracy is not properly tied to player skill. This is caused by the fact that not only does the slider head have it's own judgement, but so does the slider repeat, each slider tick, and the slider as a whole (the tail also has a judgement, but this doesn't affect accuracy as mentioned above). This means that every slider gives out at least 2 judgements affecting accuracy, and sometimes many, many more, which overweights it. So in a map that is 50% sliders and 50% circles, at least 67% of your accuracy will come from your performance on the sliders, and if there are hold sliders or long sliders this could easily be as high as 75% or maybe even 80%. This behavior is abusable as well because an increased slider tick rate will result in much increased accuracy for exactly the same performance. Further, kick sliders with many repeats could be used to greatly increase accuracy. For example, imagine I made a jump map with 200 circles and 100 sliders, where 10 of these sliders are hold sliders with 7 repeats each and 10 others are low velocity sliders with 45 ticks (I checked an actual map and this is reasonable at the max tick rate). In this theoretical map, there are 300 clickable objects, 100 slider completion scores, and 520 slider components, that all equally affect accuracy. This means more than _**half**_ of the accuracy on this map would come from easy to hit, unclicked objects that never give partial accuracy. If a player got 90% accuracy from the clickable objects and slider completions, but hit all of the easy to hit slider ticks and repeats, they would end up with 95.65% accuracy, which is a very significant boost. Maybe this scenario is exaggerated (though I tried to be within the bounds of reason), but similar scenarios could certainly exist to a lesser extent and mappers could strategically utilize this to inflate accuracy on farm maps, if they wanted to do so.

The way I see it only clickable objects should have an effect on accuracy, as they are the only objects that actually require any timing; however, I understand your desire to account for accuracy in following a slider (otherwise, what's the point of sliders), but this honestly doesn't seem like a good way of doing it to me. The way in stable was ok, but I understand why you might want to move to a system that holds the player more accountable for their ability to follow sliders. This a bit of a hard problem to solve without overweighting sliders, but I have a couple of ideas. The first is to get rid of slider ticks and repeats having judgements and to divide the judgements for the head and slider completion in half (that way they add to 300). The problem with this is it isn't the easiest to reconcile with the current 300, 100, 50, 0 judgement system. My second suggestion is to get rid of all of the judgements that come from the slider except the completion judgement, while modifying it's formula. Perhaps to this:
```
if (hitFraction == 1 && HeadCircle.Result.Type == HitResult.Great)  
    r.Type = HitResult.Great;
else if (hitFraction >= 0.5 && HeadCircle.Result.Type >= HitResult.Good)
    r.Type = HitResult.Good;
else if (HeadCircle.Result.Type >= HitResult.Meh)
    r.Type = HitResult.Meh;
else
    r.Type = HitResult.Miss;
```
IDK if either of these solutions would be good, but I'm sure there are many other possible solutions to the issue. At the very least slider ticks and repeats should not have a direct affect on accuracy, or at least much less weighting than clickable objects.

This got a little long & ranty, and it is maybe not the best organized. Sorry. I hope I made some sense.

**Comment by Volleo6144** [comment]
I just wanted to post here: The example code shown near the bottom of the issue text would issue a Miss judgment for the slider if the player misses the head circle, even if they follow the rest of the slider (and I often miss the head of a slider then hit the rest of it for a 100... most of my combo breaks on stable would turn into misses instead of Goods...)

**Comment by Fvern** [comment]
Something has changed since 2018, because the sliderend judgements do affect accuracy, but somehow both the head and the tail don't affect combo?
![osu_2020-03-19_13-37-02](https://user-images.githubusercontent.com/25032854/77035138-ccf19380-69e6-11ea-89cf-7196fbf95719.jpg)
![screenshot615](https://user-images.githubusercontent.com/25032854/77035142-d1b64780-69e6-11ea-9470-5cc50c93aace.jpg)
I FCed this map twice on each engine (with ScoreV2 on stable), mistiming the first slider and SSing the rest. On lazer I got 2 100s on the first slider, compared to stable's 1, and somehow got 2 less combo. Viewing the stable replay on lazer causes the 2 100s, lower acc and combo, until the result screen displays the score I got on stable, which of course is just taken from the file itself and not the play.
As the person who opened the issue mentioned, if this is intentional it's really bad design, as you get 2 bad judgements from mistiming one hitobject. Sliderticks and repeats are bad as well, because being able to get 500 300s and 6 misses on a single minute-long slider when you only clicked 1 object is just wrong.
I would suggest making the behaviour in line with stable's ScoreV2.
